### ISSUE TITLE
Fetch function not working when specifying a Request body class

### ISSUE THREAD

**Issue by big-mike-88** [oven-sh/bun#9846]
### What version of Bun is running?

1.1.0

### What platform is your computer?

Microsoft Windows NT 10.0.19044.0 x64

### What steps can reproduce the bug?

1. Create a basic reproducer following the svelte kit steps provided in the bun docs: https://bun.sh/guides/ecosystem/sveltekit
2. Create a +page.server.ts file in /src/routes like so:

```ts
import type { PageServerLoad } from './$types';

export const load = (async () => {
    console.log('normal fetch')
    console.log(await (await fetch('https://httpbin.org/headers', {
        headers: {
            'Authorization': 'bearer asdf'
        }
    })).json())

    console.log('fetch request body')
    const req = new Request('https://httpbin.org/headers');

    req.headers.set('Authorization', 'bearer asdf');
    console.log(req);

    console.log(await (await fetch(req)).json());
}) satisfies PageServerLoad
```

3. Access http://localhost:5173 in the browser. Check the server logs.

### What is the expected behavior?

I should see no difference in calling https://httpbin.org/headers when using a fetch request with an `init` versus one with `new Request()`. Both of them should be like so:

```js
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Authorization: "bearer asdf",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.0",
    "X-Amzn-Trace-Id": "Root=1-660bcd05-757ce29925d759ef5a10d20a",
  },
}
```

### What do you see instead?

I see that the header info is not kept when specifying a request object:

normal fetch
```js
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Authorization: "bearer asdf",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.0",
    "X-Amzn-Trace-Id": "Root=1-660bcd05-757ce29925d759ef5a10d20a",
  },
}
```

fetch request body
```
Request (0 KB) {
  method: "GET",
  url: "https://httpbin.org/headers",
  headers: Headers {
    "authorization": "bearer asdf",
  }
}
```

```js
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.0",
    "X-Amzn-Trace-Id": "Root=1-660bcd05-692971c71361804c753988df",
  },
}
```

Notice how `Authorization: "bearer asdf"` is missing from the second response output, the only difference being that I use a `new Request()` instead of the fetch config object

### Additional information

This completely kneecaps sveltekit hooks, because the fetch hook that intercepts fetch calls relies on modifying the request body object using the `Request` class. A lot of other stuff breaks, so I suspect that no configuration on the Request class survives, because I keep getting 405 method not allowed from other libraries, meaning that not even the method information is kept.

Also, using the provided server fetch from sveltekit cannot be configured at all. Calling:
```ts
import type { PageServerLoad } from './$types';

export const load = (async ({fetch}) => {
    console.log('normal fetch')
    console.log(await (await fetch('https://httpbin.org/headers', {
        headers: {
            'Authorization': 'bearer asdf'
        }
    })).json())
    
}) satisfies PageServerLoad
```
Returns this, which also doesn't contain the authorization header
```js
{                                                                 
  headers: {                                                      
    Accept: "*/*",                                                
    "Accept-Encoding": "gzip, deflate, br",                       
    Host: "httpbin.org",                                          
    "User-Agent": "Bun/1.1.0",                                    
    "X-Amzn-Trace-Id": "Root=1-660bcf6f-3f9daaef1f2a004f7b365edd",
  },                                                              
}         
```

**Comment by big-mike-88** [comment]
Small update, making a basic project with bun init and then bun run on the same conde (minus the svelte load, of course) works:

normal fetch
```js
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Authorization: "bearer asdf",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.0",
    "X-Amzn-Trace-Id": "Root=1-660d428e-72b541e24ce82ca82563dee1",
  },
}
```
fetch request body
```
Request (0 KB) {
  method: "GET",
  url: "https://httpbin.org/headers",
  headers: Headers {
    "authorization": "bearer asdf",
  }
}
```
```js
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Authorization: "bearer asdf",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.0",
    "X-Amzn-Trace-Id": "Root=1-660d428e-7b0304a87d58b22e2b04a872",
  },
}
```

This is probably svelte specific. I'll try to play around with it some more.

**Comment by big-mike-88** [comment]
After trying various things, what worked is using `bun run dev` instead of `bun --bun run dev`. Not sure if this is a performance hit or not. I'm not sure if this is a solution or a sloppy workaround. In any case, the docs specify to use --bun and that it "should" work like that.

**Comment by Septimus** [comment]
I seem to be having a similar issue when using `--bun`. I believe this is needed in order to run the vite dev server with bun instead of node.

There is definitely something going on with fetch with bun.

**Comment by big-mike-88** [comment]
As long as the prod version works without issue (and it does) I'm not too attached with vite dev being on node for the time being, however this definitely should be fixed. I'll try on my mac later this week to see if the issue persists, however I doubt it since such a big thing would have probably been noticed by now

**Comment by big-mike-88** [comment]
Ok tested on Mac as well:

  VITE v5.2.8  ready in 314 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
normal fetch
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Authorization: "bearer asdf",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.2",
    "X-Amzn-Trace-Id": "Root=1-6611593a-764ff9d37438ce1355df52a6",
  },
}
fetch request body
Request (0 KB) {
  method: "GET",
  url: "https://httpbin.org/headers",
  headers: Headers {
    "authorization": "bearer asdf",
  }
}
{
  headers: {
    Accept: "*/*",
    "Accept-Encoding": "gzip, deflate, br",
    Host: "httpbin.org",
    "User-Agent": "Bun/1.1.2",
    "X-Amzn-Trace-Id": "Root=1-6611593b-04c71e2e70641a746e01ccf9",
  },
}

It actually doesn't work here as well. Seems to be a svelte-kit error in general, not a windows error.

**Comment by big-mike-88** [comment]
Closing this in favor of #4718

I don't know how I didn't stumble upon it sooner but no point in having this be a duplicate issue. I'll comment on there.

**Comment by Jarred-Sumner** [comment]
I think https://github.com/oven-sh/bun/pull/10969 will fix this

It seems like SvelteKit is overriding the `headers` property when overriding the `Request` subclass, and previously we were not handling that case (similar to method and url, except `headers` this time)

We are still not handling `redirect` or `signal`

**Comment by Jarred-Sumner** [comment]
This is fixed in more recent versions of Bun (> 1.1.13?)

```ts
❯ cat fetchy.js
       │ File: fetchy.js
   1   │ class MYRequest extends Request {
   2   │
   3   │ };
   4   │
   5   │ const req = new MYRequest("https://example.com");
   6   │ Object.defineProperty(req, "headers", {value: { "X-A": "123"}});
   7   │
   8   │ await fetch(req, {verbose: true});


❯ bun fetchy.js
[fetch] > HTTP/1.1 GET https://example.com/
[fetch] > x-a: 123
[fetch] > Connection: keep-alive
[fetch] > User-Agent: Bun/1.1.18
[fetch] > Accept: */*
[fetch] > Host: example.com
[fetch] > Accept-Encoding: gzip, deflate, br

[fetch] < 200 OK
[fetch] < Content-Encoding: gzip
[fetch] < Accept-Ranges: bytes
[fetch] < Age: 483560
[fetch] < Cache-Control: max-age=604800
[fetch] < Content-Type: text/html; charset=UTF-8
[fetch] < Date: Mon, 01 Jul 2024 20:53:29 GMT
[fetch] < Etag: "3147526947"
[fetch] < Expires: Mon, 08 Jul 2024 20:53:29 GMT
[fetch] < Last-Modified: Thu, 17 Oct 2019 07:18:26 GMT
[fetch] < Server: ECAcc (sac/2530)
[fetch] < Vary: Accept-Encoding
[fetch] < X-Cache: HIT
```
